dashboard/login.py

```diff
@@ -47,6 +47,14 @@ def sign_up_form() -> None:
                     st.error(
                         "A user with this email already exists. Please log in instead.")
 
+                    # Add new user
+                    if add_new_user_to_database(conn, new_email_input, username):
+                        st.success(
+                            "Successfully signed up! You can now log in.")
+                        conn.close()
+                        # After successful signup, show login form again
+                        st.session_state.show_signup = False
+                        st.rerun()
```

Comment:
This is actually different from main. I'll add suggestion of what I think it's supposed to be.
```suggestion
                # Add new user
                else:
                    insert_user(conn, username, new_email_input,
                                new_password_input)
                    st.success(
                        "Successfully signed up! Redirecting to login page...")
                    conn.close()
                    # After successful signup, show login form again
                    time.sleep(2)
                    st.session_state.show_signup = False
                    st.rerun()
```